model-optimizer/extensions/ops/tensor_iterator.py

```diff
@@ -372,6 +372,10 @@ def check_field(record, field):
                 in_rec_end = in_shape[in_rec['axis']] if not check_field(in_rec, 'end') else \
                     in_shape[in_rec['axis']] + 1 + in_rec['end']
                 in_rec_start = in_rec['start']
+            elif check_field(in_rec, 'end') and in_rec['end'] < 0 and \
+                    check_field(in_rec, 'start') and in_rec['start'] < 0:
+                in_rec_end = in_rec['end']
+                in_rec_start = in_rec['start']
             else:
                 in_rec_end = ti_node.in_port(in_rec['external_port_id']).data.get_shape()[in_rec['axis']]
                 in_rec_start = 0
```

Comment:
I propose on line #392 to check that `in_rec_end` and `in_rec_start` are not `dynamic_value` before iteration count computation because they can be retrieved from shapes.